apps/console/src/features/users/components/guests/pages/invitee-role-selection.tsx

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IdentifiableComponentInterface, RolesInterface, TestableComponentInterface } from "@wso2is/core/models";
+import {
+    DocumentationLink,
+    EmphasizedSegment,
+    Heading,
+    LinkButton,
+    PrimaryButton,
+    useDocumentation
+} from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Divider, Form, Grid, Modal } from "semantic-ui-react";
+import { UserInviteInterface } from "../../../../../extensions/components/admin-developer/models";
+import { RoleType } from "../models/invite";
+
+interface InviteeRoleSelectionPropsInterface extends IdentifiableComponentInterface, TestableComponentInterface {
+    invitee: UserInviteInterface;
+    showSelectionModal: boolean;
+    handleSelectionModalClose: () => void;
+    handleInviteeRolesUpdate: (inviteeID: string, RoleList: string[]) => void;
+    readOnly: boolean;
+    /**
+     * Specifies if the form is submitting.
+     */
+    isSubmitting?: boolean;
+}
+
+/**
+ * Invitee role selection component.
+ *
+ * @param  props - Props injected to the component.
+ *
+  */
+export const InviteeRoleSelection: FunctionComponent<InviteeRoleSelectionPropsInterface> = (
+    props: InviteeRoleSelectionPropsInterface
+): ReactElement => {
+
+    const {
+        invitee,
+        showSelectionModal,
+        handleSelectionModalClose,
+        handleInviteeRolesUpdate,
+        readOnly,
+        isSubmitting
+    } = props;
+
+    const { t } = useTranslation();
+    const { getLink } = useDocumentation();
+
+    const [ roleList ] = useState<RolesInterface[]>(undefined);
```

Comment:
Don't we need a set `roleList` function ? If continue this way this will always be undefined.